maven-core/src/main/java/org/apache/maven/plugin/internal/AbstractMavenPluginParametersValidator.java

```diff
@@ -0,0 +1,153 @@
+package org.apache.maven.plugin.internal;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.maven.plugin.descriptor.Parameter;
+import org.apache.maven.shared.utils.logging.MessageBuilder;
+import org.apache.maven.shared.utils.logging.MessageUtils;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluator;
+import org.codehaus.plexus.configuration.PlexusConfiguration;
+import org.slf4j.Logger;
+
+/**
+ * Common implementations for plugin parameters configuration validation.
+ *
+ * @author Slawomir Jaranowski
+ */
+abstract class AbstractMavenPluginParametersValidator implements MavenPluginConfigurationValidator
+{
+
+    // plugin author can provide @Parameter( property = "session" ) in this case property will always evaluate
+    // so, we need ignore those
+
+    // source org.apache.maven.plugin.PluginParameterExpressionEvaluator
+    private static final List<String> IGNORED_PROPERTY_VALUES = Arrays.asList(
+        "basedir",
+        "executedProject",
+        "localRepository",
+        "mojo",
+        "mojoExecution",
+        "plugin",
+        "project",
+        "reactorProjects",
+        "session",
+        "settings"
+    );
+
+    private static final List<String> IGNORED_PROPERTY_PREFIX = Arrays.asList(
+        "mojo.",
+        "plugin.",
+        "pom.",
```

Comment:
The `pom` prefix has already been removed, no? Why ignore?

Reply:
still exist in `PluginParameterExpressionEvaluator`
https://github.com/apache/maven/blob/master/maven-core/src/main/java/org/apache/maven/plugin/PluginParameterExpressionEvaluator.java#L245

I can remove it here if there is plan to remove such prefix from maven ... IHMO should be done together with change in `PluginParameterExpressionEvaluator`

Reply:
Add a note with the JIRA issue where it has been removed in Maven 4. @mthmulders Did remove it.

Reply:
I see one about prefixless https://issues.apache.org/jira/browse/MNG-7404

Reply:
Beat me, but I am certain that we have worked on it because `pom.` is deprecated.

Reply:
b2a21f12f8a0598dd4a286177e340bda0148e9ba

Reply:
https://issues.apache.org/jira/browse/MNG-7244

Reply:
ok, I will remove `pom` for 4.0.x ... but restore it in 3.9.x